app/services/import/legacy/transform-contacts.service.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+/**
+ * Transforms NALD company data into a valid object that matches the WRLS structure
+ * @module ImportLegacyTransformContactsService
+ */
+
+const FetchContactsService = require('./fetch-contacts.service.js')
+const ContactPresenter = require('../../../presenters/import/legacy/contact.presenter.js')
+const CompanyContactPresenter = require('../../../presenters/import/legacy/company-contact.presenter.js')
+const ImportContactValidator = require('../../../validators/import/contact.validator.js')
+
+/**
+ * Transforms NALD company data into a validated object that matches the WRLS structure
+ *
+ * @param {string} regionCode - The NALD region code
+ * @param {string} licenceId - The NALD licence ID
+ * @param {object[]} transformedCompanies
+ *
+ * @returns {Promise<object>} an object representing an array of valid WRLS transformed companies and
+ * an array of companies from the db
+ */
+async function go (regionCode, licenceId, transformedCompanies) {
+  const naldContacts = await FetchContactsService.go(regionCode, licenceId)
+
+  naldContacts.forEach((naldContact) => {
+    const matchingCompany = _matchingCompany(transformedCompanies, naldContact)
+
+    const transformedContact = ContactPresenter.go(naldContact)
+
+    ImportContactValidator.go(transformedContact)
+
+    matchingCompany.contact = { ...transformedContact, dataSource: 'nald' }
+
+    if (naldContact.licence_role_id && naldContact.start_date) {
```

Comment:
We inner join to WRLS licence roles, so if there wasn't a match we'd have no result rather than a blank `licence_role_id`. And I don't think there are any NALD licence versions with no `EFF_ST_DATE` else I suspect `to_date()` would blow up and cause the query to crash.

So, I don't think we need this check here, but I could be wrong? 🤷

Reply:
No you are correct. 

It would have been needed in the previous version. But it now redundant. 

Good spot